packages/upward-js/lib/resolvers/ServiceResolver.js

```diff
@@ -101,9 +101,12 @@ class ServiceResolver extends AbstractResolver {
                     return { data };
                 }
             })
-            .catch(e => {
+            .catch(async e => {
+                const { print } = require('graphql/language/printer');
                 throw new Error(
-                    `ServiceResolver: Request to ${endpoint.toString()} failed: ${e}`
+                    `ServiceResolver: Request to ${endpoint.toString()} failed: ${e}.\n\nQuery: ${print(
+                        await parsedQuery.render()
+                    )}\n\nVariables:\n\n ${JSON.stringify(variables)}`
```

Comment:
For other viewers: These enhancements aren't directly related to the PageBuilder PoC work, but they are reviewed and tested and they aren't specific to PageBuilder. This work was initially branched off a PoC branch for Adobe Experience Manager integration, and during debugging we made these changes to help with debugging.